Tasks/PublishSymbols/Publish-Symbols.ps1

```diff
@@ -0,0 +1,284 @@
+    {
+        Write-Host "StatusCode '$($_.Exception.Response.StatusCode)' returned on account $symbolServiceUri"
+        
+        if ($_.Exception.Response.StatusCode -eq 404)
+        {
+            throw "The VSTS Symbol Server feature is not enabled for this account. See https://go.microsoft.com/fwlink/?linkid=846265 for instructions on how to enable it.`n`n "
+        }
+        
+        throw  $_
+    }
+    $versionHeader = $versionResponse.Headers["symbol-client-version"]
+
+    "Most recent version is $($versionHeader)" | Write-Verbose
+
+    return $versionHeader
+}
+
+function Publish-Symbols([string]$symbolServiceUri, [string]$requestName, [string]$sourcePath, [string]$expirationInDays, [string]$personalAccessToken)
+{
+    "Using endpoint $($symbolServiceUri) to create request $($requestName) with content in $($sourcePath)" | Write-Verbose
+
+    # the latest symbol.app.buildtask.zip and use the assemblies in it.
+    $assemblyPath = Update-SymbolClient $SymbolServiceUri $env:Temp
+
+    # Publish the files
+    try
+    {
+        if ( $sourcePath ) 
+        { 
+            $sourcePath = $sourcePath.TrimEnd("\")
+            "Removing trailing '\' in SourcePath. New value: $($sourcePath)" | Write-Verbose
+        }
+    
+        $args = "publish --service `"$symbolServiceUri`" --name `"$requestName`" --directory `"$sourcePath`"" 
+
+        if ( $expirationInDays )
+        {
+             $args  += " --expirationInDays `"$expirationInDays`""
+        }
+
+        if ( $personalAccessToken )
+        {
+             $args  += " --patAuth `"$personalAccessToken`""
+        }
+        else
+        {
+             $args += " --aadAuth"
+        }
+
+        $publishResult = Run-SymbolCommand $assemblyPath $args
+
+        if ( $publishResult )
+        {
+            "Service assigned request ID: $($publishResult.Id)" | Write-Host
+            "Request will automatically expire on $($publishResult.ExpirationDate) unless manually extended." | Write-Host
+            return $publishResult.Id
+        }
+    }
+    catch [Exception]
+    {
+        # throw $_ here improves the error message from being
+        #    Exception calling "GetResult" with "0" argument(s): <actual interesting exception>
+        # to just
+        #    <actual interesting exception>
+        # which I find more clear
+        throw  $_
+    }
+}
+
+function Run-SymbolCommand([string]$assemblyPath, [string]$arguments)
+{
+    # Require tracelevel verbose to get JSON output
+    $qarg = ""
+    if ($arguments.EndsWith("-q"))
+    {
+        $qarg = "-q"
+        $arguments = $arguments.Substring(0, $arguments.Length - 2)
+    }
+
+    $exe = "$assemblyPath\symbol.exe"
+    $arguments += " --tracelevel verbose " + $qarg
+
+    $displayArgs = $arguments | ForEach-Object { $_ -replace "--patAuth [^ ]+","--patAuth (auth)" }
+    Write-Host "Running: $exe $displayArgs"
+
+    $processInfo = New-Object System.Diagnostics.ProcessStartInfo
```

Comment:
This is going to make the task appear hung until the task completes. If you use Invoke-VstsTool you will get the output interactively and the function handles throwing if exit code is non-zero and also prints the header `##[command]<your_command_line` (appears distinguished in web UI) so it's transparent to the end user what the task is doing.